controllers/machine.go

```diff
@@ -171,12 +168,16 @@ func (mrc *machineReconcileContext) Reconcile() error {
 	return nil
 }
 
-func (mrc *machineReconcileContext) patchHardware(hw *tinkv1.Hardware) error {
+// patchStates patches a hardware's metadata and instance states.
+func (mrc *machineReconcileContext) patchStates(hw *tinkv1.Hardware, mdState, iState string) error {
```

Comment:
nit: Can we name this `patchHardwareStates`? patchStates sounds a bit generic, since we are specifically patching hardware here.